tests/e2e-kubernetes/testsuites/cache.go

```diff
@@ -0,0 +1,188 @@
+		for _, f := range l.cleanup {
+			errs = append(errs, f(ctx))
+		}
+		framework.ExpectNoError(errors.NewAggregate(errs), "while cleanup resource")
+	}
+	BeforeEach(func(ctx context.Context) {
+		l = local{}
+		l.config = driver.PrepareTest(ctx, f)
+		DeferCleanup(cleanup)
+	})
+
+	createPod := func(ctx context.Context, additionalMountOptions []string, podModifiers ...func(*v1.Pod)) *v1.Pod {
+		cacheDir := randomCacheDir()
+
+		vol := createVolumeResourceWithMountOptions(ctx, l.config, pattern, append(additionalMountOptions, fmt.Sprintf("cache %s", cacheDir)))
+		deferCleanup(vol.CleanupResource)
+
+		pod := e2epod.MakePod(f.Namespace.Name, nil, []*v1.PersistentVolumeClaim{vol.Pvc}, admissionapi.LevelPrivileged, "")
+		ensureCacheDirExistsInNode(pod, cacheDir)
+		for _, pm := range podModifiers {
+			pm(pod)
+		}
+
+		pod, err := createPod(ctx, f.ClientSet, f.Namespace.Name, pod)
+		framework.ExpectNoError(err)
+		deferCleanup(func(ctx context.Context) error { return e2epod.DeletePodWithWait(ctx, f.ClientSet, pod) })
+
+		return pod
+	}
+
+	Describe("Cache", func() {
+		Describe("Local", func() {
+			It("basic file operations as root", func(ctx context.Context) {
+				pod := createPod(ctx, []string{"allow-delete"}, func(pod *v1.Pod) {
+					pod.Spec.Containers[0].SecurityContext.RunAsUser = ptr.To(root)
+					pod.Spec.Containers[0].SecurityContext.RunAsGroup = ptr.To(root)
+				})
+				checkBasicFileOperations(f, pod, e2epod.VolumeMountPath1)
+			})
+
+			It("basic file operations as non-root", func(ctx context.Context) {
+				mountOptions := []string{
+					"allow-delete",
+					"allow-other",
+					fmt.Sprintf("uid=%d", *e2epod.GetDefaultNonRootUser()),
+					fmt.Sprintf("gid=%d", defaultNonRootGroup),
+				}
+				pod := createPod(ctx, mountOptions, func(pod *v1.Pod) {
+					pod.Spec.Containers[0].SecurityContext.RunAsUser = e2epod.GetDefaultNonRootUser()
+					pod.Spec.Containers[0].SecurityContext.RunAsGroup = ptr.To(defaultNonRootGroup)
+					pod.Spec.Containers[0].SecurityContext.RunAsNonRoot = ptr.To(true)
+				})
+
+				checkBasicFileOperations(f, pod, e2epod.VolumeMountPath1)
+			})
+
+			It("two containers in the same pod using the same cache", func(ctx context.Context) {
+				testFile := filepath.Join(e2epod.VolumeMountPath1, "helloworld.txt")
+
+				pod := createPod(ctx, []string{"allow-delete"}, func(pod *v1.Pod) {
+					// Make it init container to ensure it runs before regular containers
+					pod.Spec.InitContainers = append(pod.Spec.InitContainers, v1.Container{
+						Name:  "populate-cache",
+						Image: e2epod.GetDefaultTestImage(),
+						Command: e2epod.GenerateScriptCmd(
+							fmt.Sprintf("echo 'hello world!' > %s", testFile)),
+						VolumeMounts: []v1.VolumeMount{
+							{
+								Name:      volumeName1,
+								MountPath: e2epod.VolumeMountPath1,
+							},
+						},
+					})
+				})
+
+				e2evolume.VerifyExecInPodSucceed(f, pod, fmt.Sprintf("cat %s | grep -q 'hello world!'", testFile))
```

Comment:
I don't understand what feature this is trying to test

Reply:
This test-case verifies multiple containers can use the same cache/Mountpoint without any problem. The init container writes to `testFile` and this container verifies it has the expected content.

Reply:
Good point! Should be pre-populated with https://github.com/awslabs/mountpoint-s3-csi-driver/pull/292/commits/70310991b4f191f10eb0672ee2278d26a2d67811